### PR TITLE
Create cloudfoundry.md

### PR DESCRIPTION
**Pull request by PeteLevineA**
## Hey, I just made a Pull Request!

Create a deployment target of Cloud Foundry for the backstage documentation.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Hey Team!
The team that maintains the Cloud Foundry CLI and other various projects was hoping to get in a document on how to deploy backstage to open source cloud foundry. https://www.cloudfoundry.org/ I hope I have followed the other documents examples okay enough. I'm happy to refer more to documentation if you feel some of the code is too verbose etc. Thank you!

### CODE REVIEW

**Comment by xandroc** [comment]
Hello @freben can you take a look at this PR, thanks

**Review by xandroc — APPROVED** [review]

File: docs/deployment/cloudfoundry.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+cp -v \
+  yarn.lock  \
+  package.json  \
+  packages/backend/dist/skeleton.tar.gz  \
+  packages/backend/dist/bundle.tar.gz  \
+  app-config.yaml \
+  $BUILD_DIR
+```
+
+Since backstage uses yarn berry (> v1.x.x) and the buildpack for Cloud Foundry only supports v1.x.x and below we need to create a new .yarnrc file in the build directory
```
Comment (vinzscam) [inline docs/deployment/cloudfoundry.md:63]: I think some of these steps need to be updated as `@backstage/create-app` now automatically creates a `.yarnrc` file

**Review by vinzscam — CHANGES_REQUESTED** [review]

File: docs/deployment/cloudfoundry.md (hunk 1 above)
Comment (xandroc) [reply docs/deployment/cloudfoundry.md]: Hi @vinzscam, `@backstage/create-app` does generate a `.yarnrc.yml` file compatible with yarn 2+, but I didn't see a `.yarnrc` file in my run of create-app. In this case, creating a `.yarnrc` file is meant to support yarn v1.x

Comment (vinzscam) [reply docs/deployment/cloudfoundry.md]: ah makes sense, sorry for the confusion!

Comment (vinzscam) [reply docs/deployment/cloudfoundry.md]: Does the `.yarnrc.yml` file need to be manually removed?

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

I think it'd be best to add this guide to https://github.com/backstage/backstage/tree/master/contrib instead. We strive to keep documentation in `docs` up to date, but this isn't documentation that we'll be able to maintain over time, so I think the best home for it is in contrib

**Pull request closed without merging**